packages/apollo-engine-reporting/src/extension.ts

```diff
@@ -104,32 +105,20 @@ export class EngineReportingExtension<TContext = any>
       host: null,
       path: null,
     });
-    if (this.options.privateHeaders !== true) {
-      for (const [key, value] of o.request.headers) {
-        if (
-          this.options.privateHeaders &&
-          Array.isArray(this.options.privateHeaders) &&
-          // We assume that most users only have a few private headers, or will
-          // just set privateHeaders to true; we can change this linear-time
-          // operation if it causes real performance issues.
-          this.options.privateHeaders.some(privateHeader => {
-            // Headers are case-insensitive, and should be compared as such.
-            return privateHeader.toLowerCase() === key.toLowerCase();
-          })
-        ) {
-          continue;
-        }
-
-        switch (key) {
-          case 'authorization':
-          case 'cookie':
-          case 'set-cookie':
-            break;
-          default:
-            this.trace.http!.requestHeaders![key] = new Trace.HTTP.Values({
-              value: [value],
-            });
-        }
+
+    if (this.options.sendHeaders || this.options.privateHeaders != null) {
+      if (this.options.sendHeaders) {
+        makeHTTPRequestHeaders(
+          this.trace.http,
+          o.request.headers,
+          this.options.sendHeaders,
+        );
+      } else if (this.options.privateHeaders != null) {
+        makeHTTPRequestHeadersLegacy(
```

Comment:
What if instead of having a separate legacy function, we just changed the EngineReportingAgent constructor to look for the old options, and if they are passed in, to:

- throw an error immediately if the new option is passed to
- otherwise, delete the old option from `this.options` and set the new option instead.

Reply:
Sounds reasonable. I made a separate legacy function because it was a little more straightforward to write tests, but this would be cleaner!